### ISSUE TITLE
Warnings when building with GCC 9

### ISSUE THREAD

**Issue by elliottslaughter** [StanfordLegion/Legion-SNAP#2]
My attempts to upgrade the Legion CI have revealed that SNAP generates warnings in newer GCC versions. A sample of the warning message is below.

```
g++-9 -MMD -o /builds/StanfordLegion/legion/runtime/legion/legion_c.cc.o -c /builds/StanfordLegion/legion/runtime/legion/legion_c.cc  -DSNAP_USE_RELAXED_COHERENCE  -std=c++11  -fabi-version=0 -std=c++11  -march=native -fopenmp -DUSE_HDF -I/usr/include/hdf5/serial -O0 -ggdb  -Wall -Werror  -I/builds/StanfordLegion/legion/runtime -I/builds/StanfordLegion/legion/runtime -I/builds/StanfordLegion/legion/runtime/mappers 
sweep.cc: In static member function ‘static void MiniKBATask::sse_implementation(const Task*, const std::vector<Legion::PhysicalRegion>&, Context, Runtime*)’:
sweep.cc:674:25: error: ignoring attributes on template argument ‘__m128d’ {aka ‘__vector(2) double’} [-Werror=ignored-attributes]
  674 |     AccessorRO<__m128d,3> fa_qim;
      |                         ^
```

The relevant part of this is `ignoring attributes on template argument`. Based on what I can tell from reading the relevant issues in [GCC](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=97222) and [Clang](https://github.com/llvm/llvm-project/issues/47018), this is an *intentional* behavior that has very surprising consequences. In particular, it means that the type `__vector(2) double` (this is what `__m1238d` expands to) gets treated as `double`. That is, the `__vector(2)` gets stripped because it's not "part of the type" according to the explanations in the linked bug reports.

In short, when passing a vector into a template, it gets treated not as a vector at all but as a scalar. This seems like a pretty big deal.

The workaround that's suggested online is to wrap the `__m128d` in a struct:

```c++
struct s__m128d {
  __m128d x;
};
```

I just want to make sure this is an acceptable solution before I put any effort into writing up a PR for this.

Thanks.

**Comment by elliottslaughter** [comment]
I've looked a bit more at the code, and it looks like the code mainly uses the accessor to get a pointer, which it then casts to a `__m128d *__restrict__`. Assuming this is always how the data is accessed, it may actually be fine to ignore the warning, in which case I can shut it off. But we'd better be sure we never call a `.read()` or `.write()` method on the accessor.

**Comment by lightsighter** [comment]
Stupid compiler writers, types should be types, if they've macro-ed them for their own convenience that is their own damn problem. 

Feel free to modify SNAP however you best see fit to address the issue.

**Comment by elliottslaughter** [comment]
I am attempting to work around via ecd89179899d19872e346bd769f38558005d7724.
